src/components/Modal/README.md

```diff
@@ -258,70 +212,425 @@ export default {
+		},
+	},
+};
+</script>
+```
+
+_MutliViewDemoModal.vue_
+
+```vue
+<template>
+	<m-modal>
+		<m-transition-resize>
+			<component
+				:is="currentView"
+				:switch-view="switchView"
+			/>
+		</m-transition-resize>
+	</m-modal>
 </template>
 
 <script>
-import MyModal from './MyModal.vue';
+import { MModal } from '@square/maker/components/Modal';
+import { MTransitionResize } from '@square/maker/components/TransitionResize';
+import MutliViewDemoFirstView from 'doc/MutliViewDemoFirstView.vue';
+import MutliViewDemoSecondView from 'doc/MutliViewDemoSecondView.vue';
 
 export default {
+	name: 'MutliViewDemoModal',
+
 	components: {
-		MyModal,
+		MModal,
+		MTransitionResize,
 	},
 
 	data() {
 		return {
-			isModalOpen: false,
+			boolean: true,
 		};
 	},
+	computed: {
+		currentView() {
+			if (this.boolean) {
+				return MutliViewDemoFirstView;
+			}
+			return MutliViewDemoSecondView;
+		},
+	},
+
+	methods: {
+		switchView() {
+			this.boolean = !this.boolean;
+		},
+	},
+};
+</script>
+```
+
+_MutliViewDemoFirstView.vue_
+
+```vue
+<template>
+	<div>
+		<img
+			class="cover-photo"
+			src="https://picsum.photos/800/300"
+		>
+		<br>
+		<m-modal-content>
+			<m-heading>
+				First view heading
+			</m-heading>
+			<m-text>
+				First view content
+			</m-text>
+		</m-modal-content>
+
+		<m-responsive-action-bar>
+			<m-action-bar-button
+				key="close"
+				shape="pill"
+				color="#f6f6f6"
+				@click="modalApi.close()"
+			>
+				<x-icon class="icon" />
+			</m-action-bar-button>
+			<m-action-bar-button
+				key="confirm"
+				shape="pill"
+				full-width
+				@click="switchView"
+			>
+				Switch to second view
+			</m-action-bar-button>
+		</m-responsive-action-bar>
+	</div>
+</template>
+
+<script>
+import { MHeading } from '@square/maker/components/Heading';
+import { MText } from '@square/maker/components/Text';
+import { MModalContent, modalApi } from '@square/maker/components/Modal';
+import { MResponsiveActionBar, MActionBarButton } from '@square/maker/components/ActionBar';
+import XIcon from '@square/maker-icons/X';
+
+export default {
+	name: 'MutliViewDemoFirstView',
+
+	components: {
+		MHeading,
+		MText,
+		MModalContent,
+		MResponsiveActionBar,
+		MActionBarButton,
+		XIcon,
+	},
+
+	inject: {
+		modalApi,
+	},
+
+	props: {
+		switchView: {
+			type: Function,
+			required: true,
+		},
+	},
+};
+</script>
+
+<style scoped>
+.cover-photo {
+	width: 100%;
+	height: 300px;
+	object-fit: cover;
+	object-position: center;
+}
+
+.icon {
+	width: 24px;
+	height: 24px;
+}
+</style>
+```
+
+_MutliViewDemoSecondView.vue_
+
+```vue
+<template>
+	<div>
+		<img
+			class="cover-photo"
+			src="https://picsum.photos/600/600"
+		>
+		<br>
+		<m-modal-content>
+			<m-heading>
+				Second view heading
+			</m-heading>
+			<m-text>
+				Second view content
+			</m-text>
+		</m-modal-content>
+
+		<m-responsive-action-bar>
+			<m-action-bar-button
+				key="close"
+				shape="pill"
+				color="#f6f6f6"
+				@click="modalApi.close()"
```

Comment:
This ends up closing the second view and first view. This should only close the current view, in this case the SecondView and reveal the previous view, FirstView

Reply:
This is not a bug, this works as intended, but since it sounds like you want it to behave more like a "flow" I've updated the example.